### ISSUE TITLE
Allow configuration for IPv4 binding

### ISSUE THREAD

**Issue by Sytten** [ory/x#769]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/x/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/x/blob/master/CONTRIBUTING.md).
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

Golang by default binds to IPv6, which doesn't work for everybody.
It would be nice if we could specify `tcp4` for:
https://github.com/ory/x/blob/6511eb1be9fe8389e93b50955a88121c1b728b22/networkx/listener.go#L30

### Describe your ideal solution

Ideally the parent service config can pass down an override for the IPv4 vs IPv6 binding with a default to IPv6.
I can push the PR, but I am not sure if we want to offer another function to avoid a breaking change in the interface.

```golang
func MakeListener(network string, address string, socketPermission *configx.UnixPermission) (net.Listener, error)
```

### Workarounds or alternatives

There is none right now.

### Version

Kratos 1.1.0

**Comment by alnr** [comment]
This should be possible by specifying an IPv4 literal as the listening address, no?

Please reopen if I'm mistaken.

**Comment by Sytten** [comment]
You are mistaken, by default it will always use IPv6 even with an IPv4 address is given at least for common cases like 0.0.0.0

**Comment by alnr** [comment]
Confirmed, and very weird. Go listens on IPv6 given `net.Listen("tcp", "0.0.0.0:0")`.

We can't change any interfaces.

On my machine, `net.Listen("tcp", "127.0.0.1:0")` binds a IPv4 address. So the problem is probably only acute for you want to listen on all IPv4 addresses (which is not currently possible)?

We could add a special case just for "0.0.0.0" and force IPv4 in that case. WDYT?

**Comment by Sytten** [comment]
Not stale

**Comment by alnr** [comment]
Would the solution outlined [here](https://github.com/ory/x/issues/769#issuecomment-2027099096) work for you?
